pallets/allocations/src/migrations.rs

```diff
@@ -0,0 +1,145 @@
+				let storage_item2_prefix: &[u8] = b"CoinsConsumed";
+
+				if have_storage_value(pallet_prefix, storage_item1_prefix, &[])
+					&& have_storage_value(pallet_prefix, storage_item2_prefix, &[])
+				{
+					remove_storage_prefix(pallet_prefix, storage_item1_prefix, &[]);
+					remove_storage_prefix(pallet_prefix, storage_item2_prefix, &[]);
+
+					<StorageVersion<T>>::put(crate::Releases::V2_0_21);
+
+					log::info!(
+						"on_runtime_upgrade[{:#?}]=> Removed Oracles & CoinsConsumed, Migrated to storage version {:?}",
+						line!(),
+						<StorageVersion<T>>::get()
+					);
+				} else {
+					panic!(
+						"on_runtime_upgrade[{:#?}]=> Oracles & CoinsConsumed doesn't exist",
+						line!()
+					);
+				}
+
+				T::DbWeight::get().reads_writes(2, 2)
+			} else {
+				log::info!(
+					"on_runtime_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+					line!(),
+				);
+				T::DbWeight::get().reads(1)
+			}
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			use frame_support::storage::migration::get_storage_value;
+
+			log::info!(
+				"pre_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+				line!(),
+				crate::Releases::V2_0_21,
+				<StorageVersion<T>>::get(),
+			);
+
+			if <StorageVersion<T>>::get() == Releases::V0_0_0Legacy {
+				let pallet_prefix: &[u8] = b"Allocations";
+				let storage_item1_prefix: &[u8] = b"Oracles";
+
+				let maybe_stored_data =
+					get_storage_value::<Vec<T::AccountId>>(pallet_prefix, storage_item1_prefix, &[]);
+
+				if let Some(stored_data) = maybe_stored_data {
+					log::info!(
+						"pre_upgrade[{:#?}]=> Oracles count :: [{:#?}]",
+						line!(),
+						stored_data.len(),
+					);
+				} else {
+					log::info!(
+						"pre_upgrade[{:#?}]=> Oracles is_none :: [{:#?}]",
+						line!(),
+						maybe_stored_data.is_none(),
+					);
+				}
+			} else {
+				log::info!(
+					"pre_upgrade[{:#?}]=> Version mismatch. This probably should be removed",
+					line!(),
+				);
+			}
+
+			Ok(())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			use frame_support::storage::migration::get_storage_value;
+			log::info!(
+				"post_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+				line!(),
+				crate::Releases::V2_0_21,
+				<StorageVersion<T>>::get(),
+			);
+
+			if <StorageVersion<T>>::get() == Releases::V2_0_21 {
+				let pallet_prefix: &[u8] = b"Allocations";
+				let storage_item1_prefix: &[u8] = b"Oracles";
+				let storage_item2_prefix: &[u8] = b"CoinsConsumed";
+
+				assert!(get_storage_value::<Vec<T::AccountId>>(pallet_prefix, storage_item1_prefix, &[]).is_none());
+				assert!(get_storage_value::<BalanceOf<T>>(pallet_prefix, storage_item2_prefix, &[]).is_none());
+			} else {
+				log::info!(
+					"post_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+					line!(),
+				);
+			}
+
+			Ok(())
+		}
+	}
+}
```

Comment:
Some tests to make the bot happy here and to give us confidence the try-runtime will detect problem with a runtime upgrade we want it to detect.

Reply:
Sure

Reply:
Here the try-runtime feature which adds the pre and post upgrade functions is intended to be the test for the upgrade. However codecove it's an internal logic that codecove is not aware. It would be nice If we can silent codecove for this particular file. Otherwise we can just create a ticket to do it properly for all our pallet migrations later.

Reply:
Just saw you have used `#![cfg_attr(tarpaulin, ignore)]` for your other PR, is it what we want here too?

Reply:
@simonsso could you take up @aliXsed  question regarding `#![cfg_attr(tarpaulin, ignore)]`